src/local.rs

```diff
@@ -163,32 +156,60 @@ pub fn kill_network(node_registry: &NodeRegistry, keep_directories: bool) -> Res
     Ok(())
 }
 
+pub struct LocalNetworkOptions {
+    pub faucet_bin_path: PathBuf,
+    pub join: bool,
+    pub node_count: Option<u16>,
+    pub peers: Option<Vec<Multiaddr>>,
+    pub safenode_bin_path: PathBuf,
+    pub skip_validation: bool,
+}
+
 pub async fn run_network(
     node_registry: &mut NodeRegistry,
-    safenode_bin_path: &Path,
-    faucet_bin_path: &Path,
-    node_count: Option<u16>,
-    skip_validate: bool,
+    service_control: &dyn ServiceControl,
+    network_options: LocalNetworkOptions,
 ) -> Result<()> {
     let launcher = LocalSafeLauncher {
-        safenode_bin_path: safenode_bin_path.to_path_buf(),
-        faucet_bin_path: faucet_bin_path.to_path_buf(),
+        safenode_bin_path: network_options.safenode_bin_path.to_path_buf(),
+        faucet_bin_path: network_options.faucet_bin_path.to_path_buf(),
     };
 
-    let mut port = 12000;
-    let mut rpc_port = 13000;
-    let rpc_client = RpcClient::new(&format!("https://127.0.0.1:{rpc_port}"));
-    let genesis_multiaddr =
-        run_node(port, rpc_port, None, &launcher, node_registry, &rpc_client).await?;
+    let peers = if network_options.join {
+        if let Some(peers) = network_options.peers {
+            peers
+        } else {
+            let peer = node_registry
+                .nodes
+                .iter()
+                .find(|n| n.get_multiaddr().is_some())
+                .ok_or_else(|| eyre!("Unable to obtain a peer to connect to"))?;
+            vec![peer.get_multiaddr().unwrap()]
```

Comment:
We can use `find_map` instead of `find` here.
```suggestion
            let multi_addr = node_registry
                .nodes
                .iter()
                .find_map(|n| n.get_multiaddr())
                .ok_or_else(|| eyre!("Unable to obtain a peer to connect to"))?;
            vec![multi_addr]
```

Reply:
Good spot!